include/aspect/stokes_matrix_free.h

```diff
@@ -118,6 +130,31 @@ namespace aspect
          */
         double pressure_scaling;
 
+        /**
+         * Table which stores the product of viscosity derivative with respect to pressure
+         * and newton derivative scaling factor.
+         */
+        const Table<2, VectorizedArray<number>> *viscosity_derivative_wrt_pressure_table;
+
+        /**
+         * Table which stores the strain rate for eacc cell.
+         */
+        const Table<2, SymmetricTensor<2, dim, VectorizedArray<number>>>
+        *strain_rate_table;
+
+        /**
+         * Table which stores the product of the follow three variables:
+         * viscosity derivative with respect to strain rate,
+         * newton derivative scaling factor, and alpha.
+         */
+        const Table<2, SymmetricTensor<2, dim, VectorizedArray<number>>>
+        *viscosity_derivative_wrt_strain_rate_table;
+
+        /**
+          * Information on the stabilization.
+          */
```

Comment:
Could you extend this? How is `is_symmetric` related to the stabilization? I suppose it shows if the matrix is symmetric, and based on that you need to use a different type of stabilization in the assembly of the Newton matrix? If so, please add.